ledger/partial/ptrie/node.go

```diff
@@ -28,13 +28,19 @@ type node struct {
 // newNode creates a new node with the provided value and no children
 func newNode(hashValue []byte, height int) *node {
 	n := new(node)
-	n.hashValue = hashValue
+	n.hashValue = make([]byte, 0, common.HashLen)
+	n.hashValue = append(n.hashValue, hashValue...)
+	// pad the hash to Hashlen
+	paddingLen := common.HashLen - len(hashValue)
+	n.hashValue = append(n.hashValue, common.EmptyHash[:paddingLen]...)
```

Comment:
same here, how much faster is this code in compare to the simple assignment?

Reply:
This specific function isolated became slower. However, the idea is that every node has an allocated slice, that will be passed to the computing functions as the result slice, the computation functions would write in ] the node slice directly, instead of allocating other temporary slices.

Reply:
I haven't looked at the partial trie optimizations yet, I mainly changed this to be consistent with the complete tries updates (since the computation functions are shared). I'll create a ticket for partial tries